### PR TITLE
Sort flags alphabetically

### PR DESCRIPTION
**Pull request by wiardvanrij**
Signed-off-by: Wiard van Rij <wiard@outlook.com>

Relates to https://github.com/thanos-io/thanos/issues/4034

It sorts the flags alphabetical

* [x] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

The latest release of the kingpin package is from 17 Dec 2017. It was missing any method to do some funky stuff to get this done. That is, unless we want to manually order all flags in the code. So;

* Replaced v2.2.6 of kingpin with latest commit on their master branch so `UsageFuncs` becomes available
* Used the default template with an extra function call before the flag output 
* Added a function that does a sort on flag name

I think this might fall under "it's not stupid if it works" but this change felt nasty lol. Open for review anyway, LMK.

## Verification

Locally tested. 
Will update further if this is the way to go...

### CODE REVIEW

File: pkg/extkingpin/app.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ package extkingpin
 import (
 	"fmt"
 	"os"
+	"sort"
+	"text/template"
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply pkg/extkingpin/app.go:10]: We kind of want text ;p

File: pkg/extkingpin/app.go [hunk 2]
```diff
@@ -39,6 +87,13 @@ type App struct {
 // NewApp returns new App.
 func NewApp(app *kingpin.Application) *App {
 	app.HelpFlag.Short('h')
+	app.UsageTemplate(UsageTemplate)
+	app.UsageFuncs(template.FuncMap{
+		"alphabetical": func(data []*kingpin.FlagModel) []*kingpin.FlagModel {
```
Comment (bwplotka) [inline pkg/extkingpin/app.go:92]:
```suggestion
		"alphabeticalSort": func(data []*kingpin.FlagModel) []*kingpin.FlagModel {
```

**Review by bwplotka — COMMENTED** [review]
Nice!

Can we try running `make docs` to update our docs? 🤗

**Comment by wiardvanrij** [comment]
> Nice!
> 
> Can we try running `make docs` to update our docs? 🤗

Yea, I wanted to review this first and discuss it before making it final :)

File: pkg/extkingpin/app.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (wiardvanrij) [reply pkg/extkingpin/app.go:10]: ignore

**Comment by wiardvanrij** [comment]
I don't know why the test is failing.

Local: 

```
make docs
all modules verified
>> building Thanos binary in /Users/wiard/go/bin
 >   thanos
>> generating docs
msg=success
>> cleaning white noise
```

-> 

```
 make check-docs
all modules verified
>> building Thanos binary in /Users/wiard/go/bin
 >   thanos
>> checking docs generation
msg=success
>> checking links (DISABLED for now)
```

**Comment by bwplotka** [comment]
Now all works, just flaky test, retried

**Review by bwplotka — APPROVED** [review]
Thank you! LGTM!

cc @idoqo

**Comment by wiardvanrij** [comment]
> Now all works, just flaky test, retried

It's because I cannot access /tmp locally - non-root (:  It makes up some funky /var directory. I manually fixed this (: 
Also I'm taking over your smiley behaviour..

**Comment by idoqo** [comment]
> Thank you! LGTM!
> 
> cc @idoqo

Right, looks great! Will update the docs in #3993 and see how that goes. Thanks @wiardvanrij

**Pull request merged**